pkg/loadtester/agentloadtester.go

```diff
@@ -0,0 +1,337 @@
+
+	for i := 0; i < t.params.Rooms; i++ {
+		roomName := utils.NewGuid(fmt.Sprintf("room-%d-", i))
+
+		loadTestRoom := &LoadTestRoom{
+			params: t.params,
+		}
+
+		t.lock.Lock()
+		t.testRooms[roomName] = loadTestRoom
+		t.lock.Unlock()
+
+		group.Go(func() error {
+			if err := loadTestRoom.start(roomName); err != nil {
+				log.Printf("Failed to connect to room %s: %v", roomName, err)
+				return err
+			}
+
+			_, err := loadTestRoom.publishEchoTrack()
+			if err != nil {
+				log.Printf("Failed to publish echo track to room %s: %v", roomName, err)
+				loadTestRoom.stop()
+				return err
+			}
+			loadTestRoom.stats.echoTrackPublished = true
+
+			err = loadTestRoom.dispatchAgent()
+			if err != nil {
+				log.Printf("Failed to dispatch agent to room %s: %v", roomName, err)
+				loadTestRoom.stop()
+				return err
+			}
+
+			<-groupCtx.Done()
+			log.Printf("Context cancelled for room %s, cleaning up", roomName)
+			loadTestRoom.stop()
+			return nil
+		})
+		// time.Sleep(1 * time.Second)
+	}
+	log.Printf("Agent load tester started successfully, waiting for duration: %s", t.params.Duration.String())
+
+	if err := group.Wait(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (t *AgentLoadTester) Stop() {
+	t.printStats()
+	t.lock.Lock()
+	defer t.lock.Unlock()
+
+	for roomName, room := range t.testRooms {
+		room.stop()
+		delete(t.testRooms, roomName)
+	}
+}
+
+func (r *LoadTestRoom) start(roomName string) error {
+	if r.isRunning() {
+		return nil
+	}
+
+	identity := "echo-participant"
+
+	r.room = lksdk.NewRoom(&lksdk.RoomCallback{
+		ParticipantCallback: lksdk.ParticipantCallback{
+			OnTrackSubscribed: r.onTrackSubscribed,
+			OnTrackSubscriptionFailed: func(sid string, rp *lksdk.RemoteParticipant) {
+				log.Printf("Track subscription failed, lp:%v, sid:%v, rp:%v/%v", identity, sid, rp.Identity(), rp.SID())
+			},
+		},
+		OnParticipantConnected: func(rp *lksdk.RemoteParticipant) {
+			if rp.Kind() == lksdk.ParticipantAgent {
+				r.stats.agentJoined = true
+				r.stats.agentJoinedAt = time.Now()
+			}
+		},
+		OnParticipantDisconnected: r.onParticipantDisconnected,
+	})
+
+	var err error
+	// make up to 10 reconnect attempts
+	for i := 0; i < 10; i++ {
+		err = r.room.Join(r.params.URL, lksdk.ConnectInfo{
+			APIKey:              r.params.APIKey,
+			APISecret:           r.params.APISecret,
+			RoomName:            roomName,
+			ParticipantIdentity: identity,
+		})
+		if err == nil {
+			break
+		}
+		log.Printf("Failed to join room %s (attempt %d): %v", roomName, i+1, err)
+		time.Sleep(1 * time.Second)
+	}
+	if err != nil {
+		return err
+	}
+
+	meetParticipantToken, _ := newAccessToken(r.params.APIKey, r.params.APISecret, roomName, "meet-participant")
+	r.stats.meetLink = fmt.Sprintf("https://meet.livekit.io/custom?liveKitUrl=%s&token=%s", r.params.URL, meetParticipantToken)
+	log.Printf("\nInspect the room %s in LiveKit Meet using this url: %s\n", roomName, r.stats.meetLink)
```

Comment:
this is neat! though would this print potentially 100 times if you are starting 100 agents? that'd be kind of verbose.. perhaps hide behind verbose flag?